src/async/pitfalls/pin.md

```diff
@@ -1,108 +1,108 @@
 # Pin
 
-When you await a future, you effectively move the whole stack frame from which you called `.await` to an internal data structure of your executor. If your future has pointers to data on the stack, the addresses might get invalidated. This is extremely unsafe. Therefore, you want to guarantee that the addresses your future point to don't change. That is why we need to `pin` futures. In most cases, you won't have to think about it when using futures from common libraries unless you use `select` in a loop (which is a pretty common use case). If, you implement your own future, you will likely run into this issue.
+When you await a future, you effectively move the whole stack frame from which
```

Comment:
```suggestion
When you await a future, you effectively move the whole stack frame from where
```

I guess?

Reply:
I think "from which" is more correct, but I'll rephrase.